### PR TITLE
Support Ruby 3.2 and RubyGems 3.4

### PR DESCRIPTION
**Pull request by unflxw**
Taking over from #905 (thanks @hsbt!) and changing the build matrix to test against the final Ruby 3.2 release.

### CODE REVIEW

**Comment by unflxw** [comment]
@hsbt I have not been able to reproduce the issue with loading the library from `ext/`, with JRuby and RubyGems 3.4 (using a Rails 6 app on JRuby 9.3.3.0) -- the app seems to install and load the latest AppSignal version correctly.

**Comment by hsbt** [comment]
It's only happened with `gem install appsignal`.

see Ruby 3.2 case.

```
$ ruby -v
ruby 3.2.0 (2022-12-29 revision a7d467a792) [arm64-darwin22]
$ gem i appsignal
(snip)
$ cd ~/.local/share/gem/gems/appsignal-3.3.0/
$ find . | grep bundle
./support/bundler_wrapper
./lib/appsignal_extension.bundle
```

Ruby 3.1 and RubyGems 3.3.x keep .bundle file under the ext directory.

```
$ ruby -v
ruby 3.1.4p188 (2022-12-17 revision a4ab9921a1) [arm64-darwin22]
$ gem i appsignal
(snip)
$ cd ~/.local/share/gem/gems/appsignal-3.3.0/
$ find . | grep bundle
./ext/appsignal_extension.bundle
./support/bundler_wrapper
./lib/appsignal_extension.bundle
```

File: build_matrix.yml [hunk 1]
```diff
@@ -205,12 +207,7 @@ matrix:
     - ruby: "2.7.7"
     - ruby: "3.0.4"
     - ruby: "3.1.2"
-    - ruby: "3.2.0-rc1"
-      env_vars:
-        - name: "_LIBYAML"
-          value: "1"
```
Comment (tombruijn) [inline build_matrix.yml:211]: Removing this means we can also remove these lines: https://github.com/appsignal/appsignal-ruby/blob/7f62ada8deb67a2b7d355ec0c1bc2ad1d1e2d8d1/build_matrix.yml#L28-L33

**Review by tombruijn — APPROVED** [review]

**Comment by tombruijn** [comment]
@unflxw can you add a changeset file?

**Comment by unflxw** [comment]
@tombruijn Done ✅

**Review by luismiramirez — APPROVED** [review]

**Comment by tombruijn** [comment]
@unflxw are we good to merge and ship?

**Comment by tombruijn** [comment]
This has been published in Ruby gem 3.3.1 🚀 
Thanks again!

**Pull request merged**